packages/test/test-end-to-end-tests/src/test/benchmark/DocumentMultipleDds.ts

```diff
@@ -0,0 +1,241 @@
+
+class TestDataObject extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+
+	public get _context() {
+		return this.context;
+	}
+
+	private readonly mapKey = "map";
+	public map!: SharedMap;
+
+	private readonly sharedStringKey = "sharedString";
+	public sharedString!: SharedString;
+
+	protected async initializingFirstTime() {
+		const sharedMap = SharedMap.create(this.runtime);
+		this.root.set(this.mapKey, sharedMap.handle);
+
+		const sharedString = SharedString.create(this.runtime);
+		this.root.set(this.sharedStringKey, sharedString.handle);
+	}
+
+	protected async hasInitialized() {
+		const mapHandle = this.root.get<IFluidHandle<SharedMap>>(this.mapKey);
+		assert(mapHandle !== undefined, "SharedMap not found");
+		this.map = await mapHandle.get();
+
+		const sharedStringHandle = this.root.get<IFluidHandle<SharedString>>(this.sharedStringKey);
+		assert(sharedStringHandle !== undefined, "SharedMatrix not found");
+		this.sharedString = await sharedStringHandle.get();
+	}
+}
+
+const runtimeOptions: IContainerRuntimeOptions = {
+	summaryOptions: {
+		summaryConfigOverrides: {
+			state: "disabled",
+		},
+	},
+};
+
+// implement IDocumentLoader methods
+export class DocumentMultipleDds implements IDocumentLoaderAndSummarizer {
+	private _mainContainer: IContainer | undefined;
+	private containerRuntime: ContainerRuntime | undefined;
+	private mainDataStore: TestDataObject | undefined;
+	private readonly dsCounts: number;
+	private readonly dsCountPerIteration: number;
+	private readonly _dataObjectFactory: DataObjectFactory<TestDataObject>;
+	public get dataObjectFactory() {
+		return this._dataObjectFactory;
+	}
+	private readonly runtimeFactory: ContainerRuntimeFactoryWithDefaultDataStore;
+
+	public get mainContainer(): IContainer | undefined {
+		return this._mainContainer;
+	}
+
+	public get logger(): ITelemetryLogger | undefined {
+		return this.props.logger;
+	}
+
+	private async ensureContainerConnectedWriteMode(container: IContainer): Promise<void> {
+		const resolveIfActive = (res: () => void) => {
+			if (container.deltaManager.active) {
+				res();
+			}
+		};
+		const containerConnectedHandler = (_clientId: string): void => {};
+
+		if (!container.deltaManager.active) {
+			await new Promise<void>((resolve) =>
+				container.on("connected", () => resolveIfActive(resolve)),
+			);
+			container.off("connected", containerConnectedHandler);
+		}
+	}
+
+	private async createDataStores() {
+		assert(
+			this._mainContainer !== undefined,
+			"Container should be initialized before creating data stores",
+		);
+		assert(
+			this.containerRuntime !== undefined,
+			"ContainerRuntime should be initialized before creating data stores",
+		);
+		assert(
+			this.mainDataStore !== undefined,
+			"mainDataStore should be initialized before creating data stores",
+		);
+		const totalIterations = this.dsCounts / this.dsCountPerIteration;
+		for (let i = 0; i < totalIterations; i++) {
```

Comment:
Can you add a comment here explaining why the data stores are not created together. Basically, creating them together will create too many ops which can hit the # of ops limit. This is a workaround for that.